### PR TITLE
Restore the rooms search of Element

### PR DESCRIPTION
**Pull request by Claire1817**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [X] Other :

## Content
#542 

## Screenshots / GIFs
Screenshot_1650534834

## Tested devices

- [ ] Physical
- [X] Emulator
- OS version(s):

## Checklist

- [] Changes has been tested on an Android device or Android emulator with API 21
- [X] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/tchapgouv/tchap-android-v2/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [X] Pull request updates [TCHAP_CHANGES.md](https://github.com/tchapgouv/tchap-android-v2/blob/develop/TCHAP_CHANGES.md)
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/tchapgouv/tchap-android-v2/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/tchapgouv/tchap-android-v2/blob/develop/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt)

### CODE REVIEW

**Review by yostyle — APPROVED** [review]
LGTM

File: vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt [hunk 1]
```diff
@@ -117,24 +114,6 @@ class HomeDetailFragment @Inject constructor(
         }
     }
 
-    override fun onPrepareOptionsMenu(menu: Menu) {
-        withState(viewModel) { state ->
-            val isRoomList = state.currentTab is HomeTab.RoomList
-            menu.findItem(R.id.menu_home_mark_all_as_read).isVisible = isRoomList && hasUnreadRooms
-        }
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt:124]: This code is in Element and should not be removed!

File: vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt [hunk 2]
```diff
@@ -122,17 +120,6 @@ class HomeDetailFragment @Inject constructor(
...
-        // Tchap: Update SearchView
-        // - remove max width so it can take the whole available space in landscape
-        // - show same icon as menu item
-        val searchItem = menu.findItem(R.id.menu_home_filter)
-        (searchItem?.actionView as? SearchView)?.run {
-            maxWidth = Int.MAX_VALUE
-            findViewById<ImageView>(androidx.appcompat.R.id.search_mag_icon)?.setImageDrawable(searchItem.icon)
-        }
-
-        super.onPrepareOptionsMenu(menu)
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt:135]: please also restore this call to the super method

**Review by Florian14 — APPROVED** [review]
SGTM

**Pull request merged**